mne/viz/raw.py

```diff
@@ -566,31 +566,64 @@ def _set_psd_plot_params(info, proj, picks, ax, area_mode):
             ax_list, make_label)
 
 
-def _convert_psds(psds, dB, scaling, unit, ch_names):
-    """Convert PSDs to dB (if necessary) and appropriate units."""
-    if dB:
-        where = np.where(psds.min(1) <= 0)[0]
-        if len(where) > 0:
+def _convert_psds(psds, dB, estimate, scaling, unit, ch_names):
+    """Convert PSDs to dB (if necessary) and appropriate units.
+
+    The following table summarizes the relationship between the value of
+    parameters ``dB`` and ``estimate``, and the type of plot and corresponding
+    units.
+
+    | dB    | estimate    | plot | units             |
+    |-------+-------------+------+-------------------|
+    | True  | 'power'     | PSD  | amp**2/Hz (dB)    |
+    | True  | 'amplitude' | ASD  | amp/sqrt(Hz) (dB) |
+    | True  | 'auto'      | PSD  | amp**2/Hz (dB)    |
+    | False | 'power'     | PSD  | amp**2/Hz         |
+    | False | 'amplitude' | ASD  | amp/sqrt(Hz)      |
+    | False | 'auto'      | ASD  | amp/sqrt(Hz)      |
+
+    where amp are the units corresponding to the variable, as specified by
+    ``unit``.
+    """
+    where = np.where(psds.min(1) <= 0)[0]
+    if len(where) > 0:
+        if dB:
             raise ValueError("Infinite value in PSD for channel(s) %s. "
                              "These channels might be dead." %
                              ', '.join(ch_names[ii] for ii in where))
+        else:
+            raise ValueError("Zero value in PSD for channel(s) %s. "
+                             "These channels might be dead." %
+                             ', '.join(ch_names[ii] for ii in where))
+
+    if estimate == 'auto':
+        if dB:
+             estimate = 'power'
+        else:
+             estimate = 'amplitude'
+
+    if estimate == 'amplitude':
+        np.sqrt(psds, out=psds)
+        psds *= scaling
+        ylabel = '$\\mathrm{\\frac{%s}{\\sqrt{Hz}}}$' % unit
+    else:
         psds *= scaling * scaling
+        ylabel = '$\\mathrm{%s^2}/Hz}$' % unit
+
+    if dB:
         np.log10(psds, out=psds)
         psds *= 10
-        ylabel = '%s/Hz (dB)' % unit
-    else:
-        np.sqrt(psds, out=psds)
-        psds *= scaling
-        ylabel = '$\\frac{%s}{\\sqrt{Hz}}$' % unit
+        ylabel += '$\\ \\mathrm{(dB)}$'
```

Comment:
Nice!

FYI no need to change it here (matter of taste) but in case you didn't know you can do `r'$\ \mathrm{(dB)}$` to avoid needing to double-backslash, which is often nice for these LaTeX expressions

Reply:
Looks nicer with the raw string. Just pushed the change.